### DISCUSSION TITLE
How do I call the title somewhere in the body of a book?

### DISCUSSION THREAD

**Discussion by ckunte** [typst/typst#3461] [Q&A] [answered]
I am trying to write a publications page (e.g. the second page of a book), in which the title repeats a couple of times. Given that I have already defined the `title` (e.g. "Meandering amoebas" like so, how can I recall the title without manually typing in on the publication page (`/inc/pub.typ`)?

```typst
#import "inc/template.typ": *
#show: book.with(
  title: [Meandering amoebas],
  author: "C Kunte",
  dedication: [for my teachers],
  publishing-info: [
    #include("/inc/pub.typ")
  ],
)
```

In fact, a more general question is how to return / recall a pre-defined variable, like title, author, etc?

**Comment by istudyatuni** [comment]
Define it as a separate variable, probably in the separate file. But I'm not sure, probably a better way exists. About

```typ
  publishing-info: [
    #include("/inc/pub.typ")
  ],
```

You can skip wrapping:

```typ
publishing-info: #include("/inc/pub.typ"),
```

Because `include` [returns content](https://typst.app/docs/reference/scripting/#modules)

**Reply by ckunte** [reply]
Thank you for your suggestion 🙏. Will keep this thread open for other suggestions from the community.

**Reply by PgBiel** [reply, marked as answer]
Their suggestion is correct - defining a separate variable is the best way to do this:

```typ
#let mytitle = [ABCDEF]
#show: template.with(title: mytitle, ...)
// ... later in your document ...
Title is #mytitle
```

By the way, regarding the `include` simplification suggestion, it'd be `publishing-info: include("/inc/pub.typ")` (without the hash before `include`). The hash is only used to go from markup or math mode to code mode; here you're already in code mode due to the previous hash. The hash is only necessary in the first sample because `[...]` in code mode creates a markup block, where you need to use a hash to return to code mode.

**Reply by ckunte** [reply]
Thank you for confirming. 🙏
